File: csci393/2020Spring/inclass/wk06_2020-02-19_wed/HW03_SOL.py
import time
    
def main():
    lengths = [150, 300, 450, 600, 750, 900, 1050, 1200, 1350, 1500, 1650, 1800]
    lengths += [1950, 2100, 2250, 2400, 2550, 2700, 2850, 3000, 3150, 3300]
    lengths += [3450, 3600, 3750, 3900, 4050, 4200, 4350, 4500, 4650, 4800]
    lengths += [4950, 5100, 5250, 5400, 5550, 5700, 5850, 6000]
    
    outFile = open('results.csv', 'w')
    print("n,Time", file = outFile)
    
    for length in lengths:
        print(f'length {length}')
        L = getList(length)
        
        start = time.time()
        uniqueElemets(L)
        stop = time.time()
        
        elapsed = stop - start
        print(length, elapsed, sep = ',', file = outFile)
        
    outFile.close()
    
def uniqueElemets(L):
    n = len(L);
    for i in range(n-1):
        for j in range(i+1,n):
            if L[i]  == L[j]:
                return False
    return True 

def getList(length):
    L = list(range(length))
    return L

main()

File: csci393/2020Spring/inclass/wk06_2020-02-19_wed/HW03_SOL.cpp
#include <iostream>
#include <fstream>
#include <ctime>
using namespace std;

bool uniqueElemets(int[], int);
int* getArray(int);

int main() {
	int lengths[40] = {1200, 2400, 3600, 4800, 6000, 7200, 8400, 9600, 10800,
		12000, 13200, 14400, 15600, 16800, 18000, 19200, 20400, 21600, 22800,
		24000, 25200, 26400, 27600, 28800, 30000, 31200, 32400, 33600, 34800,
		36000, 37200, 38400, 39600, 40800, 42000, 43200, 44400, 45600, 46800,
		48000};
			
	// Create output file
	ofstream outFile{};
	outFile.open("results.csv");
	
	int* array{};
	clock_t start, stop, elapsed;
	outFile << "n,Time" << endl;
	for(int i{0}; i < 40; i++){
		cout << "Length " << lengths[i] << endl;
		array = getArray(lengths[i]);
	
		start = clock();
		uniqueElemets(array, lengths[i]);
		stop = clock();
	
		delete [] array;
	
		elapsed = stop - start;
		outFile << lengths[i] << "," << elapsed << endl;
	}
	
	
	outFile.close();
	return EXIT_SUCCESS;
}

bool uniqueElemets(int array[], int length) {
	for(int i{0}; i < length - 2; i++){
		for(int j{i+1}; j < length; j++){
			if (array[i] == array[j]){
				return false;
			}
		}
	}
	return true;
}

int* getArray(int length){
	int* array = new int[length];
	for(int i{0}; i < length; i++){
		array[i] = i;
	}
	return array;
}